### PR TITLE
feat: [CDS-35470]: command flags in K8s steps

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
Adding Command Flags to Kuberenetes Steps to apply stage. This will help in passing commandflags as per need and wouldn't restrain customer to use only selected.
There may be feature flags which are not supported in client delegate kubectl so bump up the version.

## Testing
Apply
Screenshot 2022-11-30 at 10 50 51 AM (2)

Canary
Screenshot 2022-11-30 at 12 27 22 PM (2)

Rolling
Screenshot 2022-11-30 at 12 28 12 PM (2)

BlueGreen
Screenshot 2022-11-30 at 10 48 33 AM (2)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)

### CODE REVIEW

**Review by achyuth-harness — CHANGES_REQUESTED** [review]
> Reviewed 72 of 77 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 5 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/k8s/kubectl/ApplyCommand.java` line 89 at r1](https://reviewable.io/reviews/harness/harness-core/40347#-NH9Q2Sh2T-wZYDK3y6L:-NH9Q2Sh2T-wZYDK3y6M:b9rgxkf) ([raw file](https://github.com/harness/harness-core/blob/1f08475409317e450f209055c1699c2eeba882b0/960-api-services/src/main/java/io/harness/k8s/kubectl/ApplyCommand.java#L89)):*
> ```Java
>       command.append(Kubectl.flag(Flag.record));
>     }
>     if (this.commandFlags != null) {
> ```

can we use isEmpty() instead?

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/k8s/kubectl/ApplyCommand.java` line 89 at r1](https://reviewable.io/reviews/harness/harness-core/40347#-NH9Q2Sh2T-wZYDK3y6L:-NHKYgb8CSomLfOZitVr:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1f08475409317e450f209055c1699c2eeba882b0/960-api-services/src/main/java/io/harness/k8s/kubectl/ApplyCommand.java#L89)):*
<details><summary><i>Previously, achyuth-harness (Achyuth Prakash) wrote…</i></summary><blockquote>

can we use isEmpty() instead?
</blockquote></details>

Done.

**Review by vikyathharekal — CHANGES_REQUESTED** [review]
> Reviewed 76 of 77 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 2 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java` line 43 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbFAQ8Gvx7SzqjLNx:-NHMbFAQ8Gvx7SzqjLNy:b-dcoqu5) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java#L43)):*
> ```Java
>   @ApiModelProperty(dataType = SwaggerConstants.STRING_LIST_CLASSPATH)
>   ParameterField<List<TaskSelectorYaml>> delegateSelectors;
>   @Wither List<K8sStepCommandFlag> commandFlags;
> ```

Do we need this? I don't see any usage of withCommandFlags() method



_Code quote:_
```Java
@Wither
```
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryBaseStepInfo.java` line 44 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbTQS8OJ0SLPxmNVL:-NHMbTQS8OJ0SLPxmNVM:bf68xcn) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryBaseStepInfo.java#L44)):*
> ```Java
>   @ApiModelProperty(dataType = SwaggerConstants.STRING_LIST_CLASSPATH)
>   ParameterField<List<TaskSelectorYaml>> delegateSelectors;
>   @Wither List<K8sStepCommandFlag> commandFlags;
> ```

Same comment as before



_Code quote:_
```Java
 @Wither
```
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingBaseStepInfo.java` line 45 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbZSUBgEyUCDSapA4:-NHMbZSUBgEyUCDSapA5:b2bzm8a) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingBaseStepInfo.java#L45)):*
> ```Java
>   ParameterField<List<TaskSelectorYaml>> delegateSelectors;
>   @JsonIgnore String canaryStepFqn;
>   @Wither List<K8sStepCommandFlag> commandFlags;
> ```

Same comment



_Code quote:_
```Java
@Wither
```
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingRollbackBaseStepInfo.java` line 45 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbc2i67tzSSY0iNSk:-NHMbc2i67tzSSY0iNSl:b2bzm8a) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingRollbackBaseStepInfo.java#L45)):*
> ```Java
>   ParameterField<List<TaskSelectorYaml>> delegateSelectors;
>   @JsonIgnore String rollingStepFqn;
>   @Wither List<K8sStepCommandFlag> commandFlags;
> ```

Same comment



_Code quote:_
```Java
@Wither
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/AbstractK8sStepExecutorTestBase.java` line 70 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMc08f0TnFMSH1C33h:-NHMc08f0TnFMSH1C33i:b-sjhffn) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/AbstractK8sStepExecutorTestBase.java#L70)):*
> ```Java
>                           .store(storeConfig)
>                           .build();
>     List<K8sStepCommandFlag> commandFlags = asList(K8sStepCommandFlag.builder()
> ```

nit: We could use Collections.singletonList() as we have only one arg



_Code quote:_
```Java
asList
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java` line 67 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMcOY66Sdn22dzXZjy:-NHMcOY66Sdn22dzXZjz:b16q4wf) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java#L67)):*
> ```Java
>     stepParameters.setSkipSteadyStateCheck(ParameterField.createValueField(true));
>     stepParameters.setFilePaths(ParameterField.createValueField(Arrays.asList("file1.yaml", "file2.yaml")));
>     stepParameters.setCommandFlags(ParameterField.createValueField("--server-side"));
> ```

nit: Could we define a constant for this? We are using this at many places



_Code quote:_
```Java
--server-side
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java` line 93 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMd1t0AuVUctYanNr2:-NHMd1t1B0daqgSyvJyS:b94tvan) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java#L93)):*
> ```Java
>     stepParameters.setSkipSteadyStateCheck(ParameterField.ofNull());
>     stepParameters.setFilePaths(ParameterField.createValueField(Arrays.asList("file1.yaml", "file2.yaml")));
>     stepParameters.setCommandFlags(ParameterField.createValueField("--server-side"));
> ```

Looks like this test is supposed to test null parameters 



_Code quote:_
```Java
stepParameters.setCommandFlags(ParameterField.createValueField("--server-side"));
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sBGSwapServicesStepTest.java` line 102 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMdFzuDfw_4mTVX8k5:-NHMdFzuDfw_4mTVX8k6:bmcwl3z) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sBGSwapServicesStepTest.java#L102)):*
> ```Java
>   final K8sCommandFlag k8sCommandFlag =
>       K8sCommandFlag.builder().valueMap(ImmutableMap.of(K8sSubCommandType.APPLY, "--server-side")).build();
>   final List<K8sStepCommandFlag> commandFlags = asList(K8sStepCommandFlag.builder()
> ```

nit: Same comment



_Suggestion:_
```Java
asList, --server-side
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sBlueGreenStepTest.java` line 74 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMdXsX4zIgi7mtdd3w:-NHMdXsX4zIgi7mtdd3x:bmcwl3z) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sBlueGreenStepTest.java#L74)):*
> ```Java
>     K8sCommandFlag k8sCommandFlag =
>         K8sCommandFlag.builder().valueMap(ImmutableMap.of(K8sSubCommandType.APPLY, "--server-side")).build();
>     List<K8sStepCommandFlag> commandFlags = asList(K8sStepCommandFlag.builder()
> ```

nit: Same comment



_Suggestion:_
```Java
asList, --server-side
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sCanaryStepTest.java` line 84 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMe08GAhp_chlLovUV:-NHMe08GAhp_chlLovUW:bmcwl3z) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sCanaryStepTest.java#L84)):*
> ```Java
>     K8sCommandFlag k8sCommandFlag =
>         K8sCommandFlag.builder().valueMap(ImmutableMap.of(K8sSubCommandType.APPLY, "--server-side")).build();
>     List<K8sStepCommandFlag> commandFlags = asList(K8sStepCommandFlag.builder()
> ```

nit: Same comment



_Suggestion:_
```Java
asList, --server-side
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sRollingStepTest.java` line 88 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMe4STE-UI9HeCWYks:-NHMe4STE-UI9HeCWYkt:bmcwl3z) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sRollingStepTest.java#L88)):*
> ```Java
>     K8sCommandFlag k8sCommandFlag =
>         K8sCommandFlag.builder().valueMap(ImmutableMap.of(K8sSubCommandType.APPLY, "--server-side")).build();
>     List<K8sStepCommandFlag> commandFlags = asList(K8sStepCommandFlag.builder()
> ```

nit: Same comment



_Suggestion:_
```Java
asList, --server-side
```
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sStepHelperTest.java` line 4128 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMeMu67M5A4uG96tr9:-NHMeMu67M5A4uG96trA:bmcwl3z) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sStepHelperTest.java#L4128)):*
> ```Java
>   @Category(UnitTests.class)
>   public void testGetDelegateK8sCommandFlag() {
>     List<K8sStepCommandFlag> commandFlags = asList(K8sStepCommandFlag.builder()
> ```

nit: Same comment



_Suggestion:_
```Java
asList, --server-side
```
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sCommandFlagType.java` line 16 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMfsxFCb_fVDBVKHUc:-NHMfsxFCb_fVDBVKHUd:b-j76bhw) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sCommandFlagType.java#L16)):*
> ```Java
> import java.util.Set;
> import lombok.Getter;
> ```

Please add @OwnedBy
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sStepCommandFlag.java` line 28 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMfxSf-fEKJBeOXHSC:-NHMfxSf-fEKJBeOXHSD:bgpf710) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sStepCommandFlag.java#L28)):*
> ```Java
> @EqualsAndHashCode(exclude = {"flag"})
> @TypeAlias("k8sStepCommandFlag")
> @OwnedBy(CDC)
> ```

CDP? :sweat_smile: 



_Code quote:_
```Java
CDC
```
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java` line 496 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMhwHx-2DRdztp1lJX:-NHMhwHx-2DRdztp1lJY:b7wk6wc) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java#L496)):*
> ```Java
> 
>     return k8sTaskHelperBase.applyManifests(rollbackHandlerConfig.getClient(), prunedResourcesToBeRecreated,
>                k8sDelegateTaskParams, pruneLogCallback, false, null)
> ```

Why aren't we passing command flags here? :thinking: 



_Code quote:_
```Java
null
```

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 19 file(s) to review. 2 change requests. 9. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java` line 43 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbFAQ8Gvx7SzqjLNx:-NHUTlgA3kxQCNYiiIf3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java#L43)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Do we need this? I don't see any usage of withCommandFlags() method
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryBaseStepInfo.java` line 44 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbTQS8OJ0SLPxmNVL:-NHUTmQX6pViaglTeJpJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryBaseStepInfo.java#L44)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Same comment as before
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingBaseStepInfo.java` line 45 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbZSUBgEyUCDSapA4:-NHUTmvE1HTi_XdrBMVZ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingBaseStepInfo.java#L45)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Same comment
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingRollbackBaseStepInfo.java` line 45 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMbc2i67tzSSY0iNSk:-NHUTnJEDddfWaQOk423:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sRollingRollbackBaseStepInfo.java#L45)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Same comment
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java` line 93 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMd1t0AuVUctYanNr2:-NHUTvPBAyHFnmEVuEUg:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java#L93)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Looks like this test is supposed to test null parameters 
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sCommandFlagType.java` line 16 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMfsxFCb_fVDBVKHUc:-NHUU4mv3S08Q5hiClka:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sCommandFlagType.java#L16)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Please add @OwnedBy
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sStepCommandFlag.java` line 28 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMfxSf-fEKJBeOXHSC:-NHUU5bG4V3TZdDgVBh1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/K8sStepCommandFlag.java#L28)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

CDP? :sweat_smile: 
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java` line 496 at r2](https://reviewable.io/reviews/harness/harness-core/40347#-NHMhwHx-2DRdztp1lJX:-NHUU85W-P85dXckg9c7:b-ignxdp) ([raw file](https://github.com/harness/harness-core/blob/c64c600abde57c9ec175f08f46fb29ff1ffcb989/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sRollingRollbackBaseHandler.java#L496)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Why aren't we passing command flags here? :thinking: 
</blockquote></details>

I was doing changes w.r.t. request Handlers, Base handler i overlooked. Thanks for catching this, will look into it

**Review by vikyathharekal — CHANGES_REQUESTED** [review]
> Reviewed 15 of 19 files at r3, 13 of 13 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 2 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java` line 42 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwdF714sI6P0-BAAug:-NHwdF714sI6P0-BAAuh:b-zbvama) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java#L42)):*
> ```Java
>   @ApiModelProperty(dataType = SwaggerConstants.STRING_LIST_CLASSPATH)
>   ParameterField<List<TaskSelectorYaml>> delegateSelectors;
>   @YamlSchemaTypes({runtime}) List<K8sStepCommandFlag> commandFlags;
> ```

Are we allowing runtime input for this?
If yes, why not in the K8s Apply Step?
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java` line 58 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwdlH-AT8qW2oLdPDQ:-NHwdlH-AT8qW2oLdPDR:bm4frym) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java#L58)):*
> ```Java
> public class K8sApplyStepTest extends AbstractK8sStepExecutorTestBase {
>   @InjectMocks private K8sApplyStep k8sApplyStep;
>   final String commandFlag = "--server-side";
> ```

Constants could be all UPPERCASE and static too.
___
*[`125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json` line 6160 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwfixd9mMEaRyrD8K8:-NHwfixd9mMEaRyrD8K9:bzbgrx2) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json#L6160)):*
> ```JSON
>               "type": "array",
>               "items": {
>                 "$ref": "#/definitions/K8sStepCommandFlag"
> ```

Don't we have to fix BG Swap as well? 



_Code quote:_
```JSON
              "type": "array",
              "items": {
                "$ref": "#/definitions/K8sStepCommandFlag"
```
___
*[`125-cd-nextgen/src/test/resources/schema/DeploymentSteps/all.json` line 3328 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwg1YO-v5hisnjPmS-:-NHwg1YO-v5hisnjPmS0:bbxgkzp) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/test/resources/schema/DeploymentSteps/all.json#L3328)):*
> ```JSON
>               "items": {
>                 "$ref": "#/definitions/K8sStepCommandFlag"
>               }
> ```

Same comment as above



_Code quote:_
```JSON
            "commandFlags": {
              "type": "array",
              "items": {
                "$ref": "#/definitions/K8sStepCommandFlag"
              }
```

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 34 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java` line 42 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwdF714sI6P0-BAAug:-NHyaZDcBiDpgJST8OB0:b-14w2zo) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenBaseStepInfo.java#L42)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Are we allowing runtime input for this?
If yes, why not in the K8s Apply Step?
</blockquote></details>

There was an error coming in UI when it was trying to read the runtime input of ApplyStep. Once i removed the the expression for runtime input it went away. tested it with Vikrant 
___
*[`125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java` line 58 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwdlH-AT8qW2oLdPDQ:-NHyaobuAUdtYXMoLfy1:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/test/java/io/harness/cdng/k8s/K8sApplyStepTest.java#L58)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Constants could be all UPPERCASE and static too.
</blockquote></details>

Done
___
*[`125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json` line 6160 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwfixd9mMEaRyrD8K8:-NHyb0i7Bd0qczwzCrmh:b-onp26x) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json#L6160)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Don't we have to fix BG Swap as well? 
</blockquote></details>

I thought apply Manifest function was used in BGswap so i did request side changes for it but for delegate side changes i couldnt pinpoint the place for the same. We can discuss this tomorrow

**Comment by TarunUba1** [comment]
trigger smartchecks

**Comment by TarunUba1** [comment]
trigger smartchecks

**Comment by TarunUba1** [comment]
trigger codebasehash

**Review by achyuth-harness — APPROVED** [review]
> Reviewed 2 of 6 files at r5, 6 of 7 files at r7, 5 of 5 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 37 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained.

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347)* status: 37 file(s) to review. 1 change requests. 4. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/test/resources/schema/DeploymentSteps/all.json` line 3328 at r4](https://reviewable.io/reviews/harness/harness-core/40347#-NHwg1YO-v5hisnjPmS-:-NI7SP5X0WER83G8v_A3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a8438560a383253c0fbc47ccc53bea177102c8d9/125-cd-nextgen/src/test/resources/schema/DeploymentSteps/all.json#L3328)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

Same comment as above
</blockquote></details>

Done.

**Review by TarunUba1 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40347#-:-NI7SV1G7yKXmk5zdYLk:bnfp4nl)* status: 37 file(s) to review. 1 change requests. 4. unresolved discussions. 1 of 2 approvals obtained.

File: 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java [hunk 1]
```diff
@@ -1967,5 +1968,6 @@ public void register(Kryo kryo) {
     kryo.register(TerragruntRollbackTaskResponse.class, 573560);
     kryo.register(GitOpsFetchAppTaskParams.class, 573561);
     kryo.register(GitOpsFetchAppTaskResponse.class, 573562);
+    kryo.register(K8sCommandFlag.class, 601234);
```
Comment (abosii-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java:1971]: I see that we're adding new kryo registration here for the task parameters. This means that any customer that won't have updated delegate will fail with kryo exception. You can check this confluence page for more details: https://harness.atlassian.net/wiki/spaces/CDP/pages/21195653377/Delegate+Task+best+practices

**Review by abosii-harness — CHANGES_REQUESTED** [review]
Are we good with not having these changes behind FF?

File: 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java (hunk 1 above)
Comment (TarunUba1) [reply 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java:2011]: Added FF and tested these cases
https://docs.google.com/spreadsheets/d/1XL7ecZcmOOhas65rEs86fHeJgw_4D2mUcFzZvKWuVk0/edit#gid=0

**Pull request closed without merging**